src/main/java/org/folio/completablefuture/VertxFutureRepeater.java

```diff
@@ -0,0 +1,22 @@
+package org.folio.completablefuture;
+
+import java.util.function.Function;
+import java.util.function.Supplier;
+
+import io.vertx.core.Future;
+
+public final class VertxFutureRepeater {
+  /**
+   * Run <code>task</code> repeatedly until it succeeds or it has run <code>max</code> times.
+   * Return the result of the succeeding task, or the last run's failure if all runs fail.
+   */
+  public static <T> Future<T> repeat(int max, Supplier<Future<T>> task) {
```

Comment:
Is it something similar to https://vertx.io/docs/vertx-circuit-breaker/java/?

Reply:
More likely not.  This code just represents a retry strategy in case of inventory failures